### PR TITLE
Iterate sample

### PR DESCRIPTION
**Pull request by uwieder**
Iterator over Sampled  membership maps for Full Membership and Sparse Membership

### CODE REVIEW

File: platform/src/main/java/org/hillview/utils/IntSet.java [hunk 1]
```diff
@@ -97,6 +97,19 @@ public boolean add(final int k) {
         return true;
     }
 
+    /**
+     *
+     * @param pos a location in the key[] array
+     * @return in res[0] the location of the next full slot after cursor. In res[1] the value.
+     */
+    public int[] getNext(int pos) {
+        int[] res = new int[2];
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/utils/IntSet.java]: This is expensive: you allocate an array for every call.
It's better to pass the array as an out argument, or to expose the internal indexing operation and return just the internal index as an opaque value.

File: platform/src/main/java/org/hillview/utils/Randomness.java [hunk 2]
```diff
@@ -43,6 +43,21 @@ public Randomness(long seed) {
...
 
     /**
+     * @return the next pseudarandum Geometric with distribution parameter p. Should be more efficient than using logs.
+     */
+    public int nextGeometric(double p) {
+        if (p <= 0) throw new IllegalArgumentException("parameter of Geometric Distribution must be positive");
+        double sample = myPrg.nextDouble();
+        int i = 0;
+        double bar = 0;
+        while (sample > bar) {
```
Comment (mbudiu-vmw) [inline platform/src/main/java/org/hillview/utils/Randomness.java]: This is a non-constant cost too.
What is the amortized cost to do this for a full FullMembership?

File: platform/src/main/java/org/hillview/utils/IntSet.java (hunk 1 above)
Comment (uwieder) [reply platform/src/main/java/org/hillview/utils/IntSet.java]: You are right, I made the change

File: platform/src/main/java/org/hillview/utils/Randomness.java (hunk 2 above)
Comment (uwieder) [reply platform/src/main/java/org/hillview/utils/Randomness.java]: No excuse for this brain fart. I rewrote it the right way.

**Pull request merged**